Python/specialize.c

```diff
@@ -1797,21 +1798,17 @@ _Py_Specialize_BinarySubscr(
         // struct _specialization_cache):
         ht->_spec_cache.getitem = descriptor;
         ht->_spec_cache.getitem_version = version;
```

Comment:
It should make to be thread-safe by using LOCK_TYPE(), but it will be handled with separated PRs.